libraries/extractor/src/main/java/androidx/media3/extractor/text/ssa/SsaParser.java

```diff
@@ -461,11 +426,104 @@ private static Cue createCue(
       cue.setPosition(styleOverrides.position.x / screenWidth);
       cue.setLine(styleOverrides.position.y / screenHeight, LINE_TYPE_FRACTION);
     } else {
-      // TODO: Read the MarginL, MarginR and MarginV values from the Style & Dialogue lines.
       cue.setPosition(computeDefaultLineOrPosition(cue.getPositionAnchor()));
       cue.setLine(computeDefaultLineOrPosition(cue.getLineAnchor()), LINE_TYPE_FRACTION);
     }
 
+    // Apply margins if there are no overrides and we have valid positions.
+    if (styleOverrides.alignment == SsaStyle.SSA_ALIGNMENT_UNKNOWN
+        && styleOverrides.position == null
+        && cue.getPosition() != Cue.DIMEN_UNSET
+        && cue.getLine() != Cue.DIMEN_UNSET) {
+
+      // Margin from Dialogue lines takes precedence over margin from Style line.
+      float marginLeft = dialogueMarginLeft != 0f
+          ? dialogueMarginLeft / screenWidth
```

Comment:
nit: imo double-nested ternaries are quite hard to read

I would suggest unrolling this to:

```java
float marginLeft;
if (dialogueMarginLeft != 0f) {
  marginLeft = dialogueMarginLeft / screenWidth;
} else if (style != null) {
  marginLeft = style.marginLeft / screenWidth;
} else {
  marginLeft = 0f;
}
```